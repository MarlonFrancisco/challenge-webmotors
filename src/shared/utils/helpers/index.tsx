import React from "react";
import { ThemeProvider } from "styled-components";
import { theme } from "../../styled";

type WithThemeProps = {};

export const WithTheme: React.FC<WithThemeProps> = ({ children }) => (
  <ThemeProvider theme={theme}>{children}</ThemeProvider>
);
