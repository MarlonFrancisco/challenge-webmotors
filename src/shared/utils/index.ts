export type Options = {
  withoutPrefix: boolean;
  splitted: boolean;
};

export const formatCurrency = (
  value = 0,
  options?: Options
): string | string[] => {
  const formatted = new Intl.NumberFormat("pt-BR", {
    style: "currency",
    currency: "BRL",
  }).format(value);
  let apllyOptions: string | string[] = formatted;

  if (!options) {
    return formatted;
  }

  if (options.withoutPrefix) {
    apllyOptions = apllyOptions.replace("R$", "").trim();
  }

  if (options.splitted) {
    apllyOptions = apllyOptions.split(",");
  }

  return apllyOptions;
};
