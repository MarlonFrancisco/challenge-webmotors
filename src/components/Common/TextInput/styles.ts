import styled from "styled-components";
import { Theme } from "../../../shared/@types";

export const Container = styled.div`
  display: flex;
  border: 1px solid
    ${(props: { theme: Theme }) => props.theme.palette.text.primary};
  padding: 6px;
  height: 30px;
  justify-content: space-between;
`;

export const Input = styled.input`
  border: none;
  color: ${(props: { theme: Theme }) => props.theme.palette.text.secondary};
  font-weight: 600;
`;
