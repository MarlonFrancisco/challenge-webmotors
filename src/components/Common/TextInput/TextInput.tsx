import React, { ChangeEvent, ReactNode } from "react";
import { Input, Container } from "./styles";

export type TextInputProps = {
  RightComponent?: ReactNode;
  LeftComponent?: ReactNode;
  value: string;
  onChange?: (e: ChangeEvent<HTMLInputElement>) => void;
};

const TextInput: React.FC<TextInputProps> = ({
  RightComponent,
  LeftComponent,
  ...rest
}) => {
  return (
    <Container>
      {LeftComponent && LeftComponent}
      <Input {...rest} />
      {RightComponent && RightComponent}
    </Container>
  );
};

export default TextInput;
