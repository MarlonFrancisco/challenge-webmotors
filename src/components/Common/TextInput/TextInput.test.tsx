import React from "react";
import { render, fireEvent } from "@testing-library/react";
import TextInput, { TextInputProps } from "./TextInput";
import { WithTheme } from "../../../shared/utils/helpers";

const Wrapper = (props: TextInputProps) => (
  <WithTheme>
    <TextInput {...props} />
  </WithTheme>
);

describe("TextInput", () => {
  let props: TextInputProps;

  beforeEach(() => {
    props = {
      value: "webmotors",
      onChange: jest.fn(),
      LeftComponent: <p>LeftComponent</p>,
      RightComponent: <p>RightComponent</p>,
    };
  });

  it("should render without an error", () => {
    const { container } = render(<Wrapper {...props} />);
    expect(container).toMatchSnapshot();
  });

  it("should render props", () => {
    const { queryByDisplayValue, queryByText } = render(<Wrapper {...props} />);

    expect(queryByDisplayValue(props.value)).not.toBeNull();
    expect(queryByText("LeftComponent")).not.toBeNull();
    expect(queryByText("RightComponent")).not.toBeNull();
  });

  it("should call onChange function", () => {
    const { getByDisplayValue } = render(<Wrapper {...props} />);

    const inputElement = getByDisplayValue(props.value);

    const triggeredEvent = fireEvent.change(inputElement, {
      target: {
        value: "motors",
      },
    });

    expect(triggeredEvent).toBeTruthy();
    expect(props.onChange).toHaveBeenCalled();
  });
});
