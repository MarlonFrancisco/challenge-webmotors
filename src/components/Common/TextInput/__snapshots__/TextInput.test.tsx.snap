// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextInput should render without an error 1`] = `
<div>
  <div
    class="sc-bdnylx ksggrZ"
  >
    <p>
      LeftComponent
    </p>
    <input
      class="sc-gtssRu eomtQD"
      value="webmotors"
    />
    <p>
      RightComponent
    </p>
  </div>
</div>
`;
